core/lib/src/main/java/org/opennms/core/utils/PropertiesCache.java

```diff
@@ -55,17 +60,19 @@
 	private static final Logger LOG = LoggerFactory.getLogger(PropertiesCache.class);
 
     public static final String CHECK_LAST_MODIFY_STRING = "org.opennms.utils.propertiesCache.enableCheckFileModified";
+    public static final String CACHE_TIMEOUT = "org.opennms.utils.propertiesCache.cacheTimeout";
+    public static final int DEFAULT_CACHE_TIMEOUT = 600;
 
-    private static class PropertiesHolder {
+    protected static class PropertiesHolder {
         private Properties m_properties;
         private final File m_file;
         private final Lock lock = new ReentrantLock();
         private long m_lastModify = 0;
         private boolean m_checkLastModify = Boolean.getBoolean(CHECK_LAST_MODIFY_STRING);
 
-        PropertiesHolder(File file) {
-            m_file = file;
-            m_properties = null;
+        PropertiesHolder(final File file) {
+            this.m_file = file;
```

Comment:
I like the usage of "this" because it assures that fields that are referenced in an inner class are declared in this inner class. Removing an inner class field will not lead to an error if an outer class field with the same name exist. Since such a reference to a outer class field is valid neither the IDE nor the compiler will treat this as an error. This can introduce problems in the code. But of course, this is a matter of taste. I can revert these changes if you want.

Reply:
Okay, I see your point. In this case, the inner class is declared as static, so it doesn't have access to the member variables of the outer class.

Reply:
Ok, you're right. Missed the "static".